### DISCUSSION TITLE
A Concept of universal Blocking

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#6030] [Ideas]
Hey Folks,

Hello everyone,

Is anyone working on or exploring the concept of a universal block? The idea is to have a block saved in a block manager that can be updated anytime. Whenever the block is used across various templates, any updates to the block should reflect in all instances.

**Comment by ronaldohoch** [comment]
Hi, i don't know if this is it, but take a look at the Symbols: https://github.com/GrapesJS/grapesjs/discussions/4317

**Reply by glaprida** [reply]
There is a built in symbols module now, though I haven't used it.
[https://grapesjs.com/docs/guides/Symbols.html](https://grapesjs.com/docs/guides/Symbols.html#concept)

**Reply by RakulAgn** [reply]
Yes, the symbols work with the project data but not with data from another project, right?

The instance I mentioned applies not just within the same template but across all templates.

For example, if I create a symbol in Template One and use it in Template Two, and then edit the symbol, will it update in Template One? I believe it won't update automatically. However, we can check if the symbol is available in the template and update it on load, right?

**Reply by ronaldohoch** [reply]
I never used the Symbols before, so i don't know if it fits to you.
That's why i send the original discusion and not the documentation :D

**Reply by ronaldohoch** [reply]
I understood.
Here we have a version converter based on the component model's json for when there are multiple modifications that need to be applied when the template is opened

**Reply by RakulAgn** [reply]
> I understood.
> 
> Here we have a version converter based on the component model's json for when there are multiple modifications that need to be applied when the template is opened

Verison Converter ?

**Reply by RakulAgn** [reply]
> I never used the Symbols before, so i don't know if it fits to you.
> 
> That's why i send the original discusion and not the documentation :D

I went through that, but it doesn’t suit my use case. Maybe we can try some other methods and see how it’s possible.

**Reply by ronaldohoch** [reply]
> > I understood.
> > Here we have a version converter based on the component model's json for when there are multiple modifications that need to be applied when the template is opened
> 
> Verison Converter ?

It is an algorithm that compares the json from old version of a component's json to the new one, add the new options and remove the one's we don't use. I can't say more because it isn't made by me, it was @gustavohleal

**Reply by RakulAgn** [reply]
done
